File: src/workloads/ts/gc_pause.ts
const N_gc = PARAMS.inner_iters;

function step_gc_pause(): unknown {
    const t: { x: number; y: number; z: number }[] = [];
    for (let i = 0; i < N_gc; i++) {
        t[i] = { x: i, y: i * 2, z: i * 3 };
    }
    return t;
}

(globalThis as any).step = step_gc_pause;

File: src/workloads/lua/gc_pause.lua
-- gc_pause: each tick allocates inner_iters short-lived tables.
-- Host samples per-tick wall time; tail latency reflects GC behavior.

local N = PARAMS.inner_iters

function step()
    local t = {}
    for i = 1, N do
        t[i] = { x = i, y = i * 2, z = i * 3 }
    end
    return t
end
